metadata-ingestion/tests/unit/test_transform_dataset.py

```diff
@@ -2841,3 +2844,188 @@ def test_add_dataset_data_product_transformation():
     assert [item["value"]["destinationUrn"] for item in first_data_product_aspect] == [
         builder.make_dataset_urn("bigquery", "example1")
     ]
+
+
+def _test_clean_owner_urns(
+    in_pipeline_context: Any,
+    in_owners: List[str],
+    config: List[Union[re.Pattern, str]],
+    cleaned_owner_urn: List[str],
+) -> None:
+    # Return fake aspect to simulate server behaviour
+    def fake_ownership_class(entity_urn: str) -> models.OwnershipClass:
+        return models.OwnershipClass(
+            owners=[
+                models.OwnerClass(owner=owner, type=models.OwnershipTypeClass.DATAOWNER)
+                for owner in in_owners
+            ]
+        )
+
+    in_pipeline_context.graph.get_ownership = fake_ownership_class  # type: ignore
+
+    output = run_dataset_transformer_pipeline(
+        transformer_type=PatternCleanUpOwnership,
+        aspect=models.OwnershipClass(
+            owners=[
+                models.OwnerClass(owner=owner, type=models.OwnershipTypeClass.DATAOWNER)
+                for owner in in_owners
+            ]
+        ),
+        config={"pattern_for_cleanup": config},
+        pipeline_context=in_pipeline_context,
+    )
+
+    assert len(output) == 2
+    ownership_aspect = output[0].record.aspect
+    assert isinstance(ownership_aspect, OwnershipClass)
+    assert len(ownership_aspect.owners) == len(in_owners)
+
+    out_owners = [owner.owner for owner in ownership_aspect.owners]
+    assert set(out_owners) == set(cleaned_owner_urn)
+
+
+def test_clean_owner_urn_transformation(mock_datahub_graph):
+    pipeline_context = PipelineContext(run_id="transformer_pipe_line")
+    pipeline_context.graph = mock_datahub_graph(DatahubClientConfig())
+
+    user_emails = [
+        "ABCDEF:email_id@example.com",
+        "ABCDEF:123email_id@example.com",
+        "email_id@example.co.in",
+        "email_id@example.co.uk",
+        "email_test:XYZ@example.com",
+        "email_id:id1@example.com",
+        "email_id:id2@example.com",
+    ]
+
+    in_owner_urns: List[str] = []
+    for user in user_emails:
+        in_owner_urns.append(
+            builder.make_owner_urn(user, owner_type=builder.OwnerType.USER)
+        )
+
+    # remove ':ABCDEF'
+    config: List[Union[re.Pattern, str]] = [":ABCDEF"]
+    expected_user_emails: List[str] = [
+        "email_id@example.com",
+        "123email_id@example.com",
```

Comment:
Please share why pattern is `:ABCDEF`  but it is removing `ABCDEF:` from the string?

Reply:
- for input, first we are generating `user URNs` using the email ids - so the URN for `ABCDEF:123email_id@example.com` will be `urn:li:corpuser:ABCDEF:email_id@example.com`
- these URNs will be used as a input for the transformer
- so the pattern/string `:ABCDEF` will be matched against the `URN` and `URN` will be cleaned accordingly
- same like, input we are generating `user URNs` for the expected values, where the expected clean value for `ABCDEF:123email_id@example.com` is `urn:li:corpuser:email_id@example.com`

Reply:
Please see this https://github.com/datahub-project/datahub/pull/10067/files#r1528808599